اصلاح مشكلة عدم نقل مكاتب ألف الى مجلد C:/Alif3/aliflib

AhmedElTabarani opened the pull request:
<div dir = rtl>

في الوندوز عند بناء المترجم لم يقم بتحديث الملفات الخاصة بمكاتب ألف في `C:/Alif3/aliflib`   
وكان يجب القيام بنسخ المجلد `aliflib` من المترجم يدويا إلى مجلد ألف في `C:/Alif3` 
لذا قمت بإصلاح المشكلة وجعل عملية النقل عملية تلقائية اثناء البناء

AhmedElTabarani committed:
اصلاح مشكلة عدم نقل مكاتب ألف (aliflib) الى مجلد C:/Alif3/aliflib في الوندوز

AhmedElTabarani commented:
<div dir = rtl>

 بعد نقل محتوى مجلد `aliflib` الذي في المترجم الى `C:/Alif3/aliflib`
ومحاولة تشغيل اي ملف "ألف" يظهر هذا الخطأ

<div dir = ltr> 

```
alif e:\ألف\compiler\اختبارات\أكواد\الإدخال\ادخل_رقما.alif

Please report this error to the Alif Community: https://github.com/alifcommunity/compiler/issues

Alif Compiler 3.0.32 (Beta) - Windows

File : e:\ألف\compiler\اختبارات\أكواد\الإدخال\ادخل_رقما.alif
Error: In file included from 30:C:/Alif3/gcc/x86_64-w64-mingw32/include/winsock2.h:16:2: warning: #warning Please include winsock2.h before windows.h [-Wcpp] #warning Please include winsock2.h before windows.h  ^~~~~~~
```
<div dir = rtl>

المشكلة قد تكون في المكاتب الجديدة التى أضيفت وتم تحديثها
لان اي شخص كان سيحاول نقلها يدويا من غير cmake (التعديل التى قمت به للنقل تلقائيا) سيحصل معه نفس المشكلة

AhmedElTabarani committed:
windows.h و winsock2.h اصلاح خلل مكتبة

AhmedElTabarani commented:
<div dir = rtl>

هنا شرح لسبب المشكلة مع windows.h و winsock2.h

https://stackoverflow.com/questions/29069904/why-does-mingw-w64-require-winsock2-to-be-manually-included

<div dir = ltr>

> windows.h includes winsock.h by default. winsock.h and winsock2.h do not coexist. If winsock2.h is included first, it disables winsock.h, and all is good. But if winsock.h is included first, winsock2.h fails to compile, because it redeclares a lot of things that winsock.h already declares. Thus why you need to include winsock2.h first.

me00001 commented on CMakeLists.txt:
```diff
@@ -65,3 +65,13 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # Alif Compiler Source Code
 add_subdirectory(${AlifProject_SOURCE_DIR}/src Alif)
+
+# --[ Copping Files ] ----------------------------------------------------------------
+
+if (WIN32)
+    if(EXISTS "C:/Alif3/")
+
+        file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/aliflib/ DESTINATION C:/Alif3/aliflib/)
+
+    endif()
+endif()
```
<div dir="rtl">

استخدم الهدف install في CMake 

<div dir="ltr">

https://github.com/alifcommunity/compiler/blob/3d03f389ae6b1dc2708dad50b3393776cd5fc9c9/src/CMakeLists.txt#L189-L196
</div>
هذا مخصص لجنو/لينكس والأنظمة المماثلة ولكن يمكنك اعداده ليعمل على ويندوز

me00001 committed:
Create CMakeLists.txt

يمكن الأن استخدام 
mingw32-make install
في ويندوز

هذا التعديل يعطل استخدام 
cpack 
لأنه مسار مطلق
نعدل عليه حتى يقبل المسار المخصص لو ألف دعم هذا في المستبقل

me00001 committed:
Update CMakeLists.txt

استبدل ب
https://github.com/AhmedElTabarani/compiler/commit/60d88cc30a60b405d77988453693376b2121ae1c

me00001 commented:
:heart: جزاك الله خير
